### PR TITLE
fix: Fixed 10080 settings sidebar scrolling issue

### LINKED ISSUE #10080: [CAL-2151] Top banner has weird z-index position in settings
[https://github.com/calcom/cal.com/assets/8019099/10c87eb6-d4e3-4c2b-a66e-02adaa5945ad](https://github.com/calcom/cal.com/assets/8019099/10c87eb6-d4e3-4c2b-a66e-02adaa5945ad)

The banner should behave the same as on the regular pages.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2151](https://linear.app/calcom/issue/CAL-2151/top-banner-has-weird-z-index-position-in-settings)</sub>

### PR DESCRIPTION
**Pull request by bhargavtenali**
## What does this PR do?

fixes #10080 

Video Link : https://drive.google.com/file/d/1gMdWe0NhVEB6i2pXIzfTpelFNvQKGL5E/view?usp=sharing

Hi @Jaibles @PeerRich @Udit-takkar 
Hope you are doing well.
After hours of debugging to find the exact root cause. This is what I found:
Issue and Fix Summary:

1. The Issue #10080 is not because of z-index. Infact the sidebar & mainPage are having position Sticky elements with appropriate z-index. The topbar is a static element but child of Sticky Element.
2. But the issue is because we havent given max-height and top property's values for settings SideBar.
3. I had tested the same fix on mobile -> It works as expected
4. I had used React.cloneElement using which we can pass an extra prop _bannersHeight_ [top banner height] to props.SidebarContainer [settings Sidebar]
5. I had used _bannersHeight_ in props.SidebarContainer [settings Sidebar] for top and max-height
6. I feel there is no need to change anything with topBar.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

When Any topBar is present (eg: Organisation Upgrade Banner) -> Go Settings Page -> scroll through the sidebar present in Settings Page
Video Link : https://drive.google.com/file/d/1gMdWe0NhVEB6i2pXIzfTpelFNvQKGL5E/view?usp=sharing

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 1]
```diff
@@ -218,8 +220,9 @@ const SettingsSidebarContainer = ({
 
   return (
     <nav
+      style={{ maxHeight: `calc(100vh - ${bannersHeight}px)`, top: `${bannersHeight}px` }}
       className={classNames(
-        "no-scrollbar bg-muted fixed bottom-0 left-0 top-0 z-20 flex max-h-screen w-56 flex-col space-y-1 overflow-x-hidden overflow-y-scroll px-2 pb-3 transition-transform max-lg:z-10 lg:sticky lg:flex",
```
Comment (Udit-takkar) [inline packages/features/settings/layouts/SettingsLayout.tsx:222]: don't remove no-scrollbar ( not sure about using overflow-y-auto).
It causes this bug in brave browser

Screenshot 2023-07-31 at 12 07 13 PM
llbar

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 2]
```diff
@@ -502,6 +498,36 @@ export default function SettingsLayout({
...
+        bannersHeight={bannersHeight}
+      />
+    </>
+  );
+};
+
+type SidebarContainerElementProps = {
+  sideContainerOpen?: boolean;
+  bannersHeight?: number;
+  setSideContainerOpen?: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (Udit-takkar) [inline packages/features/settings/layouts/SettingsLayout.tsx:528]: this is causing type error use

setSideContainerOpen: React.Dispatch<React.SetStateAction<boolean>>;

File: packages/features/settings/layouts/SettingsLayout.tsx (hunk 1 above)
Comment (bhargavtenali) [reply packages/features/settings/layouts/SettingsLayout.tsx:222]: Hi @Udit-takkar 
We are having the same classes for main page sidebar, so brave browser issue should there as well.
Is it still ok ?

Comment (bhargavtenali) [reply packages/features/settings/layouts/SettingsLayout.tsx:222]: Did the changes now. Thanks @Udit-takkar

File: packages/features/settings/layouts/SettingsLayout.tsx (hunk 2 above)
Comment (bhargavtenali) [reply packages/features/settings/layouts/SettingsLayout.tsx]: Did the changes now. Thanks @Udit-takkar

**Review by Udit-takkar — DISMISSED** [review]
LGTM. I fixed the type error

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 3]
```diff
@@ -502,6 +498,36 @@ export default function SettingsLayout({
...
+        bannersHeight={bannersHeight}
+      />
+    </>
+  );
+};
+
+type SidebarContainerElementProps = {
+  sideContainerOpen: boolean;
+  bannersHeight?: number;
+  setSideContainerOpen: React.Dispatch<React.SetStateAction>;
```
Comment (emrysal) [inline packages/features/settings/layouts/SettingsLayout.tsx:528]:
```suggestion
  setSideContainerOpen: React.Dispatch<React.SetStateAction<boolean>>;
```

**Review by emrysal — APPROVED** [review]
Re-approving @Udit-takkar after fixing failing type error.

**Pull request merged**